src/vs/workbench/contrib/debug/browser/debugSession.ts

```diff
@@ -1231,9 +1231,13 @@ export class DebugSession implements IDebugSession, IDisposable {
 		// moment for breakpoints to set and we want to do our best to not miss
 		// anything
 		if (event.hitBreakpointIds) {
-			this.enableDependentBreakpoints(event.hitBreakpointIds);
+			this.enableDependentBreakpoints(event.hitBreakpointIds).then(() => this.runInStatusQueue(event));
```

Comment:
the status queue doesn't block the user from resuming. The queue is used for ordering events if the DA rapidly changes states, by holding F5 for example (https://github.com/microsoft/vscode/pull/197397). We actually should make sure the event gets put into the status queue synchronously.

So, we would want to change how that gets handled. But like you've done here, I would only block resuming if the event did have `event.hitBreakpointIds`, otherwise we'd need to wait for the stacktrace to come back from the DA which is not desirable; I think the possibility of missing an enablement there is preferable to the responsiveness hit to stepping.

Reply:
@connor4312 if i understand you correctly you are not suggesting to change the current solution further right ?

Reply:
I believe we should undo 217d2acf638c717420f58942fdba3831bd580d15 because it doesn't do what I think you intended to do. Instead we need a new mechanism to block resumption.

The simplest way to do this would be stashing the promise from `this.enableDependentBreakpoints(event.hitBreakpointIds)` in some `waitToResume?: Promise<void>` and `await this.waitToResume;` in the continue and step* methods of the DebugSession

Reply:
@connor4312 if you have time to fix this please go ahead. It might take some time for me to give a good solution for this to achieve what you described.